src/potions.ts

```diff
@@ -85,137 +124,217 @@ export class Potion {
-      print(
-        `Determined that ${additional} ${
-          this.potion.plural
-        } are profitable on partial embezzlers: net value ${this.net(
-          embezzlersRemaining,
-          doubleDuration
-        ).toFixed(0)}.`,
-        "blue"
-      );
-      quantityToUse += additional;
-      embezzlersRemaining = Math.max(embezzlersRemaining - additional * duration, 0);
+    values.push({
+      name: "embezzler",
+      quantity: limitFunction(embezzlerQuantity),
+      value: embezzlerValue,
+    });
+
+    // compute the number of embezzlers missed before, and their value (along with barf unless nobarf)
+    const overlapEmbezzlers = -this.overage(embezzlerTurns, embezzlerQuantity);
+
+    if (overlapEmbezzlers > 0) {
+      values.push({
+        name: "overlap",
+        quantity: limitFunction(1),
+        value: this.gross(overlapEmbezzlers, globalOptions.noBarf ? overlapEmbezzlers : undefined),
+      });
     }
-    if (embezzlersRemaining > 0 || (doubleDuration && quantityToUse > 0)) keepGoing = false;
-
-    // How many should we use with non-embezzlers?
-    if (keepGoing && this.net(0, doubleDuration) > 0) {
-      const adventureCap = estimatedTurns();
-      const tourists = adventureCap - haveEffect(this.effect()) - quantityToUse * duration;
-      if (tourists > 0) {
-        const touristQuantity = Math.ceil(tourists / duration);
-        acquire(touristQuantity, this.potion, this.gross(0, doubleDuration), false);
-        const additional = Math.min(touristQuantity, itemAmount(this.potion) - quantityToUse);
-        print(
-          `Determined that ${additional} ${
-            this.potion.plural
-          } are profitable on tourists: net value ${this.net(0, doubleDuration).toFixed(0)}.`,
-          "blue"
-        );
-        quantityToUse += additional;
+
+    const embezzlerCoverage =
+      embezzlerQuantity + (overlapEmbezzlers > 0 ? 1 : 0) * this.effectDuration();
+
+    if (!globalOptions.noBarf) {
+      // unless nobarf, compute the value of barf turns
+      // if ascending, break those turns that are not fully covered by a potion into their own value
+      const remainingTurns = Math.max(0, totalTurns - embezzlerCoverage - startingTurns);
+
+      const barfQuantity = this.usesToCover(remainingTurns, ascending);
+      values.push({ name: "barf", quantity: limitFunction(barfQuantity), value: this.gross(0) });
+
+      if (globalOptions.ascending && this.overage(remainingTurns, barfQuantity) <= 0) {
+        const ascendingTurns = Math.max(0, remainingTurns - barfQuantity * this.effectDuration());
+        values.push({
+          name: "ascending",
+          quantity: limitFunction(1),
+          value: this.gross(0, ascendingTurns),
+        });
       }
     }
 
-    if (quantityToUse > 0) {
-      if (doubleDuration) quantityToUse = 1;
-      use(quantityToUse, this.potion);
+    print(`POTION: Breakpoints for ${this.potion} (starting turns ${startingTurns}):`);
```

Comment:
Can you reduce the quantity of debug messages? Fine for debugging but I think we want less text - and more human-readable text - in the final version.